### PR TITLE
Add autoconfiguration for file share directoryclient

### PR DESCRIPTION
**Pull request by saragluna**
To resolve https://github.com/Azure/azure-sdk-for-java/issues/24413.
- Add auto-configuration for FileShare directory client.
- Change configuration name from `file-name` to `file-path` and add `directory-path`.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/storage/fileshare/AzureStorageFileShareAutoConfigurationTest.java [hunk 1]
```diff
@@ -97,6 +101,102 @@ void shareNameNotSetShouldNotConfigureShareClient() {
...
+                assertThat(context).doesNotHaveBean(ShareFileAsyncClient.class);
+            });
+    }
+
+    @Test
+    void directoryPathSetShouldConfigureDirectoryClient() {
+        ShareServiceClient shareServiceClient = mock(ShareServiceClient.class);
+        ShareClient shareClient = mock(ShareClient.class);
+        when(shareServiceClient.getShareClient("share1")).thenReturn(shareClient);
+        when(shareClient.getDirectoryClient("/directory1/directory2")).thenReturn(mock(ShareDirectoryClient.class));
```
Comment (backwind1233) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/storage/fileshare/AzureStorageFileShareAutoConfigurationTest.java:157]: I think the file-path and directory-path doesn't have `/` at the beginning.

/directory1/directory2 --> `directory1/directory2`

image

**Review by backwind1233 — APPROVED** [review]
LGTM

**Comment by backwind1233** [comment]
Do you think we should add some information in our reference doc or samples to demonstrate this?

**Comment by saragluna** [comment]
> Do you think we should add some information in our reference doc or samples to demonstrate this? @saragluna

We can add something in the Javadoc like this:
```
SMTP server host. For instance, 'smtp.example.com'.
```

**Comment by backwind1233** [comment]
> 
I don't get it.

**Pull request merged**